pkg/jobrunaggregator/jobrunaggregatorlib/util.go

```diff
@@ -157,32 +183,45 @@ type ClusterJobRunWaiter struct {
 	ProwJobMatcherFunc ProwJobMatcherFunc
 }
 
-func (w *ClusterJobRunWaiter) allProwJobsFinished(allItems []*prowv1.ProwJob) bool {
+func (w *ClusterJobRunWaiter) allProwJobsFinished(allItems []*prowv1.ProwJob) (bool, map[string]*prowv1.ProwJob) {
 	uncompletedJobMap := map[string]*prowv1.ProwJob{}
-	totalMatchedJobs := 0
+	matchedJobMap := map[string]*prowv1.ProwJob{}
+
 	for _, prowJob := range allItems {
 		if !w.ProwJobMatcherFunc(prowJob) {
 			continue
 		}
-		totalMatchedJobs++
+		jobRunID := prowJob.Labels[prowJobJobRunIDLabel]
+		matchedJobMap[jobRunID] = prowJob
 		if prowJob.Status.CompletionTime != nil {
 			continue
 		}
 
-		jobRunID := prowJob.Labels[prowJobJobRunIDLabel]
 		uncompletedJobMap[jobRunID] = prowJob
 	}
 	if len(uncompletedJobMap) == 0 {
 		logrus.Info("all jobs completed")
-		return true
+		return true, matchedJobMap
+	}
+	logrus.Infof("%d/%d jobs completed, waiting for: [%v]", len(matchedJobMap)-len(uncompletedJobMap), len(matchedJobMap), strings.Join(sets.StringKeySet(uncompletedJobMap).List(), ", "))
+	return false, matchedJobMap
+}
+
+func (w *ClusterJobRunWaiter) checkMatchedJobsForCompletion(prowJobInformer v1.ProwJobInformer) (bool, map[string]*prowv1.ProwJob, error) {
+	allItems, err := prowJobInformer.Lister().List(labels.Everything())
+	if err != nil {
+		// log and suppress the error
+		logrus.Infof("Error listing prow jobs: %v", err)
+		return false, nil, nil
```

Comment:
Ah yes, now I remember how I wound up looking at this...

pkg/jobrunaggregator/jobrunaggregatorlib/util.go:210:132: (*ClusterJobRunWaiter).checkMatchedJobsForCompletion - result 2 (error) is always nil (unparam)
func (w *ClusterJobRunWaiter) checkMatchedJobsForCompletion(prowJobInformer v1.ProwJobInformer) (bool, map[string]*prowv1.ProwJob, error) {